scripts/us_bls/cpi/generate_csv.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+Downloading and converting BLS CPI raw csv files to csv files of two columns:
+"date" and "cpi", where "date" is of the form "YYYY-MM" and "cpi" is numeric.
+
+Usage: python3 generate_csv.py
+'''
+
+
+from io import StringIO
+
+import requests
+import pandas as pd
+
+
+# Dict from series names to download links
+CSV_URLS = {
+    "cpi_u_1913_2020":
+        "https://download.bls.gov/pub/time.series/cu/cu.data.1.AllItems",
+    "cpi_w_1913_2020":
+        "https://download.bls.gov/pub/time.series/cw/cw.data.1.AllItems",
+    "c_cpi_u_1999_2020":
+        "https://download.bls.gov/pub/time.series/su/su.data.1.AllItems"
+}
+
+# Dict from series names to series IDs
+SERIES_IDS = {
+    "cpi_u_1913_2020": "CUUR0000SA0",
+    "cpi_w_1913_2020": "CWUR0000SA0",
+    "c_cpi_u_1999_2020": "SUUR0000SA0"
+}
+
+
+def main():
+    for series_name, url in CSV_URLS.items():
+        series_id = SERIES_IDS[series_name]
+        response = requests.get(url)
```

Comment:
Apparently, this can fail per https://requests.readthedocs.io/en/latest/user/quickstart/#errors-and-exceptions.  The code currently just throws the exception and thus crashes, which is fine.  It might be nice to add a comment indicating what happens upon failure.  This is a signal for a future code maintainer that error cases have been considered.

Reply:
added comment